jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/SingleSourceAllocation.kt

```diff
@@ -243,15 +243,20 @@ internal class SingleSourceAllocation(
             selectedLayers = activeLayers.filter { it.layer.height == minHeight }
 
             // This recognizes the structure used with VP9 (multiple encodings with the same resolution and unknown frame
-            // rate). In this case, we only want the low quality layer.
-            if (selectedLayers.isNotEmpty() && selectedLayers[0].layer.frameRate < 0) {
+            // rate). In this case, we only want the low quality layer. Unless we're on stage, in which case we should
+            // consider all layers.
+            if (!onStage && selectedLayers.isNotEmpty() && selectedLayers[0].layer.frameRate < 0) {
                 selectedLayers = listOf(selectedLayers[0])
```

Comment:
Do we want this to also use `minByOrNull {it.bitrate}` rather than assuming order? Or do you think that temporary bitrate inversion is more likely than encoders doing something weird?

Reply:
It's a trade-off between the risk of not sending anything even though there was a higher layer with lower bitrate available, and the risk of sending a higher quality/bitrate layer and doing an extra switch just because the bitrates were inverted when we sampled. I'd argue we should be more aggressive for on-stage, and conservative off-stage.